File: apps/api/app/schemas/summarize.py
# 网页总结的请求/草稿模型
from pydantic import BaseModel, Field


class SummarizeRequest(BaseModel):
    # 用 str + 服务端 validate_url 校验，便于给出中文 400；空串由 Pydantic 422 兜住
    url: str = Field(min_length=1)


class SummarizeDraft(BaseModel):
    url: str
    title: str
    summary: str
    suggested_tags: list[str] = []

File: apps/web/src/api/summarize.ts
import api from "./client";

// 网页总结返回的草稿（与后端 SummarizeDraft 一致；不落库）
export interface SummarizeDraft {
  url: string;
  title: string;
  summary: string;
  suggested_tags: string[];
}

// 把链接交给后端 agent 抓取+总结，返回草稿
export async function summarizeLink(url: string): Promise<SummarizeDraft> {
  const { data } = await api.post<SummarizeDraft>("/api/summarize", { url });
  return data;
}
